### PR TITLE
Add wildcard configuration for vpc name when peering VPCs

### PR DESCRIPTION
**Pull request by aoskotsky-amplify**
This adds support for the following peering syntax

``` ini
[peerings]
connection_1=*:sandbox/intranet ci/intranet
```

The above configuration would automatically peer any VPC of type `sandbox` with `ci`.

### CODE REVIEW

File: disco_aws_automation/disco_vpc.py [hunk 1]
```diff
@@ -744,68 +744,119 @@ def find_vpc_id_by_name(vpc_name):
...
+        # the source or target side might not have matched any VPCs
+        missing_vpcs = []
+        if not source_vpc_names:
+            missing_vpcs.append('%s:%s' % (source_info['vpc_name'], source_info['vpc_type']))
+        if not target_vpc_names:
+            missing_vpcs.append('%s:%s' % (target_info['vpc_name'], target_info['vpc_type']))
 
-        missing_vpcs = [vpc_name for vpc_name, vpc_object in vpc_objects.items() if not vpc_object]
         if missing_vpcs:
             logging.debug(
```
Comment (jichen-amplify) [inline disco_aws_automation/disco_vpc.py]: Maybe this should be using logging.warning()?

**Comment by pchand-amplify** [comment]
Is this bi directional?
<pre>
[peerings]
connection_1=*:[envtype]/intranet *:[envtype]/intranet
</pre>
Or allows only one way?
<pre>
[peerings]
connection_1=*:[envtype]/intranet env_name:[envtype]/intranet
</pre>

**Comment by aoskotsky-amplify** [comment]
It is bidirectional. 

This config would create a peering connection from every VPC of type `envtype` to every VPC of type `envtype2`. 
```ini
[peerings]
connection_1=*:envtype/intranet *:envtype2/intranet
```

The env type is a required field when using `*` for the env name. The following config is not allowed
```ini
[peerings]
connection_1=*/intranet */intranet
```

File: disco_aws_automation/disco_vpc.py [hunk 2]
```diff
@@ -750,68 +750,119 @@ def find_vpc_id_by_name(vpc_name):
             raise MultipleVPCsForVPCNameError("More than 1 VPC is named as {}".format(vpc_name))
 
     @staticmethod
-    def parse_peering_connection_line(line, vpc_conn):
+    # Pylint thinks this function has too many local variables
+    # pylint: disable=R0914
+    def parse_peering_connection_line(line, existing_vpcs):
```
Comment (jichen-amplify) [inline disco_aws_automation/disco_vpc.py]: This function is getting really big, probably needs to be broken down a little. One possibility is to make it just produce all the peering pairs and let the caller do the mapping to the existing VPCs. I could do that in the VPC update story too.

File: disco_aws_automation/disco_vpc.py [hunk 3]
```diff
@@ -750,68 +750,119 @@ def find_vpc_id_by_name(vpc_name):
...
+            """return a list of VPC names for VPCs that match the given vpc_name and vpc_type"""
+            return [vpc.tags['Name'] for vpc in existing_vpcs
+                    if vpc_name in ('*', vpc.tags['Name']) and vpc.tags['type'] == vpc_type]
+
+        source_info = get_endpoint_info(endpoints[0])
+        target_info = get_endpoint_info(endpoints[1])
+
+        if source_info['vpc_type'] == '*' or target_info['vpc_type'] == '*':
+            raise VPCConfigError('Wildcards are not allowed for VPC type in "%s". '
+                                 'Please specify a VPC type when using a wild card for the VPC name' % line)
```
Comment (jichen-amplify) [inline disco_aws_automation/disco_vpc.py]: This statement might be a little misleading, since we don't seem to support wild card for VPC type at all.

**Comment by pchand-amplify** [comment]
@aoskotsky-amplify ??

**Comment by bwarfield-amplify** [comment]
Maybe an opportunity to refactor DiscoVPCPeerings...

**Comment by aoskotsky-amplify** [comment]
Refactoring DiscoVPCPeerings ended up being a big change. Closing this and replacing with https://github.com/amplifylitco/asiaq/pull/173 to make reviewing easier since most of the past comments don't apply anymore.

**Pull request closed without merging**